src/query/execute_schema.c

```diff
@@ -8606,11 +8606,38 @@ do_create_entity (PARSER_CONTEXT * parser, PT_NODE * node)
   create_select = node->info.create_entity.create_select;
   if (create_select != NULL)
     {
+      DB_QUERY_TYPE *column;
+
       error = pt_get_select_query_columns (parser, create_select, &query_columns);
       if (error != NO_ERROR)
 	{
 	  goto error_exit;
 	}
+
+      /* check for mis-creating string type with -1 precision */
+      for (column = query_columns; column != NULL; column = db_query_format_next (column))
+        {
+          switch (column->domain->type->id)
```

Comment:
How about?
```
if (TP_IS_VAR_LEN_CHAR_TYPE(column->domain->type->id) &&  ( column->domain->precision == DB_DEFAULT_PRECISION || column->domain->precision > DB_MAX_VARNCHAR_PRECISION)

Reply:
I think the switch-case style is better for readability.

Reply:
Is there any reason the logic of varchar and varnchar are different?
If it's not different, the if statement I'm suggesting looks short and clear.

Reply:
the max length of VARCHAR and VARNCHAR is different and it is assigned DB_MAX_VARCHAR_PRECISION instead of DB_MAX_VARNCHAR_PRECISION at VARNCHAR type in some test cases. So I separated two cases VARCHAR and VARNCHAR.

Reply:
Okay.